### PR TITLE
[docs][ysql]Added new default roles in ysql.

### LINKED ISSUE #10444: [Docs] Add docs for newly added yugabyteDB roles
Add documentation for new roles in YugabyteDB. This applies to v2.6, 2.8, 2.9, and later.

- yb_superuser
- yb_extension

### PR DESCRIPTION
**Pull request by aishwarya24**
Resolves #10444.

- Added `yb_extension` to latest, stable, and 2.6.
- Added `yb_fdw` to latest (beta feature).
- yb_superuser is applicable only to Yugabyte Cloud and @ddhodge has added it in this [PR](https://github.com/yugabyte/yugabyte-db/pull/10844).

@netlify /latest/secure/authorization/rbac-model/

### CODE REVIEW

File: docs/content/latest/secure/authorization/rbac-model.md [hunk 1]
```diff
@@ -44,6 +44,65 @@ Roles in YSQL can represent individual users or a group of users. They encapsula
...
+```
+
+The following table describes the default YSQL roles and users in YugabyteDB clusters.
+
+| Role | Description |
+| --- | --- |
+| postgres | Superuser role used during database creation and for PostgreSQL operations. |
+| yb_extension | Role that allows non-Superuser users to create PostgreSQL extensions. |
+| yb_fdw | Role that allows non-Superuser users to [CREATE](https://www.postgresql.org/docs/11/sql-createforeigndatawrapper.html), [ALTER](https://www.postgresql.org/docs/11/sql-alterforeigndatawrapper.html), and [DROP](https://www.postgresql.org/docs/11/sql-dropforeigndatawrapper.html) the [Foreign data wrapper](https://www.postgresql.org/docs/12/ddl-foreign-data.html)(beta feature).
+| yugabyte | Superuser role used during database creation, by Yugabyte support to perform maintenance operations, and for backups (ysql_dumps).
```
Comment (polarweasel) [inline docs/content/latest/secure/authorization/rbac-model.md:70]:
```suggestion
| Role | Description |
| :--- | :---------- |
| postgres | Superuser role used during database creation and for PostgreSQL operations. |
| yb_extension | Role that allows non-Superuser users to create PostgreSQL extensions. |
| yb_fdw | Role that allows non-Superuser users to [CREATE](https://www.postgresql.org/docs/11/sql-createforeigndatawrapper.html), [ALTER](https://www.postgresql.org/docs/11/sql-alterforeigndatawrapper.html), and [DROP](https://www.postgresql.org/docs/11/sql-dropforeigndatawrapper.html) the [Foreign data wrapper](https://www.postgresql.org/docs/12/ddl-foreign-data.html)(beta feature). |
| yugabyte | Superuser role used during database creation, by Yugabyte support to perform maintenance operations, and for backups (ysql_dumps). |
```

File: docs/content/latest/secure/authorization/rbac-model.md [hunk 2]
```diff
@@ -44,6 +44,65 @@ Roles in YSQL can represent individual users or a group of users. They encapsula
...
+
+### yb_extension
+
+The `yb_extension` role allows non-Superuser roles to [create extensions](../../../api/ysql/the-sql-language/statements/ddl_create_extension/). A user granted this role can create all the extensions that are bundled in YugabyteDB.
+
+Create a role `test` and grant `yb_extension` to this role.
+
+```sql
+yugabyte=# create role test;
+yugabyte=# grant yb_extension to test ;
```
Comment (ddhodge) [inline docs/content/latest/secure/authorization/rbac-model.md:80]: remove spaces before semi colons; one below as well

**Review by ddhodge — APPROVED** [review]
LGTM

File: docs/content/latest/secure/authorization/rbac-model.md [hunk 3]
```diff
@@ -44,6 +44,66 @@ Roles in YSQL can represent individual users or a group of users. They encapsula
...
+ postgres     | Superuser, Create role, Create DB, Replication, Bypass RLS | {}
+ yb_extension | Cannot login                                               | {}
+ yb_fdw       | Cannot login                                               | {}
+ yugabyte     | Superuser, Create role, Create DB, Replication, Bypass RLS | {}
+```
+
+The following table describes the default YSQL roles and users in YugabyteDB clusters.
+<!-- Portions of this table are also under Database authorization in Yugabyte cloud -->
+
+| Role | Description |
```
Comment (sushantrmishra) [inline docs/content/latest/secure/authorization/rbac-model.md:66]: So we have any more roles available in the cluster .  Use the command  select * from pg_roles;  to display all the roles. Is the intention to document all of those roles or  just which are added specifically by Yugabyte ?

Comment (polarweasel) [reply docs/content/latest/secure/authorization/rbac-model.md:66]: @sushantrmishra I think it's sufficient to document just what we add, plus the `postgres` user (to show it's got the same attributes as `yugabyte`), so \du seems more useful here. I'll add a suggestion a few lines up to clarify the wording, though.

File: docs/content/latest/secure/authorization/rbac-model.md [hunk 4]
```diff
@@ -44,6 +44,66 @@ Roles in YSQL can represent individual users or a group of users. They encapsula
 
 * Roles inherit the privileges of all other roles granted to them.
 
+View the roles for your YugabyteDB clusters with the following command:
```
Comment (polarweasel) [inline docs/content/latest/secure/authorization/rbac-model.md:47]:
```suggestion
YugabyteDB inherits a number of roles from PostgreSQL, including the `postgres` user, and adds several new roles. View the YugabyteDB-specific roles for your clusters with the following command (or use `\duS` to display all roles):
```

File: docs/content/latest/secure/authorization/rbac-model.md [hunk 5]
```diff
@@ -44,6 +44,66 @@ Roles in YSQL can represent individual users or a group of users. They encapsula
...
+
+The following table describes the default YSQL roles and users in YugabyteDB clusters.
+<!-- Portions of this table are also under Database authorization in Yugabyte cloud -->
+
+| Role | Description |
+| :--- | :---------- |
+| postgres | Superuser role created during database creation. |
+| yb_extension | Role that allows non-Superuser users to create PostgreSQL extensions. |
+| yb_fdw | Role that allows non-Superuser users to [CREATE](https://www.postgresql.org/docs/11/sql-createforeigndatawrapper.html), [ALTER](https://www.postgresql.org/docs/11/sql-alterforeigndatawrapper.html), and [DROP](https://www.postgresql.org/docs/11/sql-dropforeigndatawrapper.html) the [Foreign data wrapper](https://www.postgresql.org/docs/12/ddl-foreign-data.html)(beta feature). |
+| yugabyte | Superuser role used during database creation, by Yugabyte support to perform maintenance operations, and for backups (ysql_dumps). |
```
Comment (polarweasel) [inline docs/content/latest/secure/authorization/rbac-model.md:71]:
```suggestion
| yugabyte | Superuser role used during database creation, by Yugabyte support to perform maintenance operations, and for backups (using ysql_dump). |
```

File: docs/content/latest/secure/authorization/rbac-model.md [hunk 6]
```diff
@@ -44,6 +44,66 @@ Roles in YSQL can represent individual users or a group of users. They encapsula
...
+
+### yb_extension
+
+The `yb_extension` role allows non-Superuser roles to [create extensions](../../../api/ysql/the-sql-language/statements/ddl_create_extension/). A user granted this role can create all the extensions that are bundled in YugabyteDB.
```
Comment (polarweasel) [inline docs/content/latest/secure/authorization/rbac-model.md:75]:
```suggestion
The `yb_extension` role allows non-superuser roles to [create extensions](../../../api/ysql/the-sql-language/statements/ddl_create_extension/). A user granted this role can create all the extensions that are bundled in YugabyteDB.
```

File: docs/content/latest/secure/authorization/rbac-model.md [hunk 7]
```diff
@@ -44,6 +44,66 @@ Roles in YSQL can represent individual users or a group of users. They encapsula
...
+```output
+ extname  | extowner | extnamespace | extrelocatable | extversion | extconfig | extcondition
+----------+----------+--------------+----------------+------------+-----------+--------------
+ pgcrypto |    16386 |         2200 | t              | 1.3        |           |
+(1 row)
+```
+
 ## Resources
 
 YSQL defines a number of specific resources, that represent underlying database objects. A resource can denote one object or a collection of objects. YSQL resources are hierarchical as described below:
```
Comment (polarweasel) [inline docs/content/latest/secure/authorization/rbac-model.md:109]:
```suggestion
YSQL defines a number of specific resources that represent underlying database objects. A resource can represent one object or a collection of objects. YSQL resources are hierarchical as described below:
```

File: docs/content/stable/secure/authorization/rbac-model.md [hunk 8]
```diff
@@ -44,6 +44,64 @@ Roles in YSQL can represent individual users or a group of users. They encapsula
 
 * Roles inherit the privileges of all other roles granted to them.
```
Comment (polarweasel) [inline docs/content/stable/secure/authorization/rbac-model.md:46]: Be sure to incorporate comments from the corresponding files above in /latest :)

File: docs/content/v2.6/secure/authorization/rbac-model.md [hunk 9]
```diff
@@ -44,6 +44,64 @@ Roles in YSQL can represent individual users or a group of users. They encapsula
 
 * Roles inherit the privileges of all other roles granted to them.
```
Comment (polarweasel) [inline docs/content/v2.6/secure/authorization/rbac-model.md:46]: Same comment as /stable... You may need to do two separate installations (2.8.0 and 2.6.7) with three separate data directories just to make extra-sure that the role lists are right, btw.

https://download.yugabyte.com/ is the easiest way to grab those all in one spot.

**Review by polarweasel — CHANGES_REQUESTED** [review]

**Review by sushantrmishra — APPROVED** [review]

**Comment by sushantrmishra** [comment]
Include the changes suggested by @polarweasel then PR looks good to me.

**Comment by aishwarya24** [comment]
@sushantrmishra - Thank you! @polarweasel - I've pushed your requested changes and here are the URL's :
https://deploy-preview-10873--infallible-bardeen-164bc9.netlify.app/latest/secure/authorization/rbac-model/
https://deploy-preview-10873--infallible-bardeen-164bc9.netlify.app/stable/secure/authorization/rbac-model/
https://deploy-preview-10873--infallible-bardeen-164bc9.netlify.app/v2.6/secure/authorization/rbac-model/

**Pull request merged**